src/imgix-core-js.js

```diff
@@ -43,15 +44,26 @@
         this.settings[key] = val;
       }
 
-      if (!Array.isArray(this.settings.domains)) {
-        this.settings.domains = [this.settings.domains];
+      if (Array.isArray(this.settings.domains)) { 
+        if (this.settings.domains.length > 1) {
+          console.warn("Warning: Domain sharding has been deprecated and will be removed in the next major version.\nAs a result, the 'domains' argument will be deprecated in favor of 'domain' instead.");
+        }
+        else if (this.settings.domains.length == 0)
+        {
+          if (typeof(this.settings.domain) != "string" && this.settings.domain != null) {
+              throw new Error('ImgixClient.settings.domain only accepts a string argument');
+          }
+          else {
+            this.settings.domains = [this.settings.domain];
+          }
+        }
       }
       else {
-        console.warn("Warning: Domain sharding has been deprecated and will be removed in the next major version.\nAs a result, the 'domains' argument will be deprecated in favor of 'domain' instead.");
+        this.settings.domains = [this.settings.domains];
       }
-
-      if (!this.settings.host && this.settings.domains.length === 0) {
-        throw new Error('ImgixClient must be passed valid domain(s)');
+      
```

Comment:
[nit] Whitespace

Reply:
I think that's fine since it's separating two different if/else blocks?

Reply:
The newline is fine, but the line should be empty.